### ISSUE TITLE
orfeo backscatter larger area -> missing band??

### ISSUE THREAD

**Issue by jdries** [Open-EO/openeo-geopyspark-driver#302]
Orfeo implementation silently surpresses errors:
```
                except RuntimeError as e:
                    logger.error(f"Error while running Orfeo toolbox. {input_tiff} {extent} EPSG {extent_epsg} {sar_calibration_lut}",exc_info=True)

```

We can use the soft errors feature to improve this.

**Comment by jdries** [comment]
yes, that's it. Not sure about the coupling: generally speaking, if the overall number of soft-errors is too high, it should fail.
Note that the most urgent requirement for me is to not tolerate any failure at all.

**Comment by jdries** [comment]
I wouldn't go deeper yet, let's roll this one out first and see if we can actually be more strict on orfeo errors.

**Comment by JeroenVerstraelen** [comment]
The implementation is currently live on development, we are waiting on the Creodias API to confirm the implementation works.

**Comment by jdries** [comment]
@JeroenVerstraelen  can this be closed?

**Comment by JeroenVerstraelen** [comment]
@jdries I believe there was a discussion after we found out that accumulators in pyspark only allow us to add to a tracker unlike spark in java where we can also read the value of a tracker at any time. At the same time we couldn't use a py4j based implementation because the BatchjobMetadataTracker class had several issues during serialization.

We decided to at least allow `max_soft_errors_ratio == 0.0 or 1.0` for now because it was urgent, so all orfeo errors are either raised or ignored. We added a TODO to read the accumulator after the collect is done and throw a error message at that point:
[Getting the value of an accumulator is not allowed in pyspark](https://github.com/Open-EO/openeo-geopyspark-driver/commit/83148d6b736136bc20a14d09a82c9afb64505915)

I can add the check + error message now if that's ok. If I am not mistaken I should be able add it completely at the end of the run_job function. That means that all output files will have been written but the user will still be notified and the batch job will be set to failed. This is not the same as the soft error system for sentinelhub where the process graph stops executing when a certain amount of errors have been reached, saving processing resources and time. But it's the best we can do with the pyspark accumulator API.
